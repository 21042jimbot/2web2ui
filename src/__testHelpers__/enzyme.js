export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
