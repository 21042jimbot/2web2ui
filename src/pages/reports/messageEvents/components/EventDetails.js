import React, { Component } from 'react';
import _ from 'lodash';
import styled from 'styled-components';
import { Box, Tooltip } from 'src/components/matchbox';
import { Panel } from 'src/components/matchbox';
import { LabelledValue, CopyField } from 'src/components';

// TODO: Replace with <Box /> when OG theme is removed
const CenteredText = styled.div`
  align-items: 'center';
`;

class EventDetails extends Component {
  static defaultProps = {
    details: {},
  };

  renderRow = ({ value, label, key }) => {
    // Renders value in a read only textfield
    if (typeof value === 'object') {
      return (
        <CenteredText as={LabelledValue} key={key} label={label}>
          <CopyField hideCopy value={JSON.stringify(value)} />
        </CenteredText>
      );
    }

    return <LabelledValue key={key} label={label} value={value} />;
  };

  // Renders key-value rows
  renderDetails = detailsToRender => {
    const { documentation } = this.props;

    return _.keys(detailsToRender).map(key => {
      // Gets tooltip content from documentation store
      const tooltipContent = _.get(documentation, [detailsToRender.type, key]);

      const row = {
        key,
        label: tooltipContent ? (
          <Tooltip dark content={tooltipContent}>
            {key}
          </Tooltip>
        ) : (
          key
        ),
        value: detailsToRender[key],
      };

      return this.renderRow(row);
    });
  };

  render() {
    // Remove formattedDate as its only used for the history table
    const detailsToRender = _.omit(this.props.details, 'formattedDate');

    return (
      <Panel.LEGACY title="Event Details">
        {/* TODO: We could stand to have a component like <Panel.Header> from Matchbox to address this scenario */}
        <Box paddingTop="500" borderBottom="400" />
        <Panel.LEGACY.Section>{this.renderDetails(detailsToRender)}</Panel.LEGACY.Section>
        <Panel.LEGACY.Section>
          <CenteredText as={LabelledValue} label="Raw Json">
            <CopyField value={JSON.stringify(detailsToRender)} />
          </CenteredText>
        </Panel.LEGACY.Section>
      </Panel.LEGACY>
    );
  }
}

export default EventDetails;
