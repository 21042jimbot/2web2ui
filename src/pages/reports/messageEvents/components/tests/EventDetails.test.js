import React from 'react';
import { shallow } from 'enzyme';
import EventDetails from '../EventDetails';

describe('EventDetails Component', () => {

  it('should render with no props', () => {
    const wrapper = shallow(<EventDetails />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with all props', () => {
    const props = {
      details: {
        key1: 'val1',
        key2: ['an', 'arrray'],
        type: 'bounce'
      },
      documentation: {
        bounce: {
          key1: 'description for key1'
        }
      }
    };

    const wrapper = shallow(<EventDetails {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
