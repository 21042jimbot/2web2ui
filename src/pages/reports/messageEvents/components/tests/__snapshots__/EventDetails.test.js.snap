// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetails Component should render with all props 1`] = `
<Panel.LEGACY
  title="Event Details"
>
  <Box
    borderBottom="400"
    paddingTop="500"
  />
  <Panel.LEGACY.Section>
    <LabelledValue
      key="key1"
      label={
        <Tooltip
          content="description for key1"
          dark={true}
        >
          key1
        </Tooltip>
      }
      value="val1"
    />
    <styled.div
      as={[Function]}
      key="key2"
      label="key2"
    >
      <CopyField
        hideCopy={true}
        value="[\\"an\\",\\"arrray\\"]"
      />
    </styled.div>
    <LabelledValue
      key="type"
      label="type"
      value="bounce"
    />
  </Panel.LEGACY.Section>
  <Panel.LEGACY.Section>
    <styled.div
      as={[Function]}
      label="Raw Json"
    >
      <CopyField
        hideCopy={false}
        value="{\\"key1\\":\\"val1\\",\\"key2\\":[\\"an\\",\\"arrray\\"],\\"type\\":\\"bounce\\"}"
      />
    </styled.div>
  </Panel.LEGACY.Section>
</Panel.LEGACY>
`;

exports[`EventDetails Component should render with no props 1`] = `
<Panel.LEGACY
  title="Event Details"
>
  <Box
    borderBottom="400"
    paddingTop="500"
  />
  <Panel.LEGACY.Section />
  <Panel.LEGACY.Section>
    <styled.div
      as={[Function]}
      label="Raw Json"
    >
      <CopyField
        hideCopy={false}
        value="{}"
      />
    </styled.div>
  </Panel.LEGACY.Section>
</Panel.LEGACY>
`;
